xgb.go

```diff
@@ -314,6 +333,15 @@ type request struct {
 // In all likelihood, you should be able to copy and paste with some minor
 // edits the generated code for the request you want to issue.
 func (c *Conn) NewRequest(buf []byte, cookie *Cookie) {
+	select {
+	case <-c.done:
```

Comment:
`NewRequest` waits until `sendRequests` processes current request. However, if an unrecoverable error happens, all goroutines, including `sendRequests` are stopped, and `NewRequest` will block forever. This select statement checks whether this connection was closed.

...however, now that you mention it, I can see a race condition here. The select statement should be:
```
select {
case <-c.done:
    return
case <-seq:
    return
}
```